pkg/plugin/plugin_test.go

```diff
@@ -65,12 +64,21 @@ func TestMerge(t *testing.T) {
 	}
 
 	armTemplate = testPluginRun(p, newCluster, oldCluster, t)
-	if strings.Contains(string(armTemplate), "\"type\": \"Microsoft.Network/networkSecurityGroups\"") {
+	if hasResourceType(armTemplate, "Microsoft.Network/networkSecurityGroups") {
 		t.Fatalf("networkSecurityGroups should not be applied during cluster upgrade")
 	}
 }
 
-func testPluginRun(p api.Plugin, newCluster *api.OpenShiftManagedCluster, oldCluster *api.OpenShiftManagedCluster, t *testing.T) (armTemplate []byte) {
+func hasResourceType(armTemplate map[string]interface{}, resType string) bool {
+	for _, res := range armTemplate["resources"].([]interface{}) {
```

Comment:
fyi - I implemented a subset of jsonpath which can be useful for this kind of thing - take a look at our use of pkg/jsonpath